File: Examples/test-suite/perl5/using_composition_runme.pl
use strict;
use warnings;
use Test::More tests => 11;
BEGIN { use_ok('using_composition') }
require_ok('using_composition');

my $f = using_composition::FooBar->new();
is($f->blah(3), 3, "FooBar::blah(int)");
is($f->blah(3.5), 3.5, "FooBar::blah(double)");
is($f->blah("hello"), "hello", "FooBar::blah(char *)");

$f = using_composition::FooBar2->new();
is($f->blah(3), 3, "FooBar2::blah(int)");
is($f->blah(3.5), 3.5, "FooBar2::blah(double)");
is($f->blah("hello"), "hello", "FooBar2::blah(char *)");

$f = using_composition::FooBar3->new();
is($f->blah(3), 3, "FooBar3::blah(int)");
is($f->blah(3.5), 3.5, "FooBar3::blah(double)");
is($f->blah("hello"), "hello", "FooBar3::blah(char *)");

File: Examples/test-suite/lua/using_composition_runme.lua
local v=require("using_composition")

local f = v.FooBar()
assert(f:blah(3) == 3, "FooBar::blah(int)")
-- Lua uses number for both float and integer
-- lua can not differentiate, C/C++ must differentiate function names
-- assert(f:blah(3.5) == 3.5, "FooBar::blah(double)")
assert(f:blah("hello") == "hello", "FooBar::blah(char *)")

f = v.FooBar2()
assert(f:blah(3) == 3, "FooBar2::blah(int)")
-- Lua uses number for both float and integer
-- Lua can not differentiate, C/C++ must differentiate function names
-- assert(f:blah(3.5) == 3.5, "FooBar2::blah(double)")
assert(f:blah("hello") == "hello", "FooBar2::blah(char *)")

f = v.FooBar3()
assert(f:blah(3) == 3, "FooBar3::blah(int)")
-- Lua uses number for both float and integer
-- Lua can not differentiate, C/C++ must differentiate function names
-- assert(f:blah(3.5) == 3.5, "FooBar3::blah(double)")
assert(f:blah("hello") == "hello", "FooBar3::blah(char *)")
